compiler/resolution/functionResolution.cpp

```diff
@@ -4334,15 +4334,33 @@ static void findVisibleFunctionsAndCandidates(
 
     handleTaskIntentArgs(info, fn);
 
-  } else {
-    findVisibleFunctions(info, visibleFns);
+    trimVisibleCandidates(info, mostApplicable, visibleFns);
+
+    findVisibleCandidates(info, mostApplicable, candidates);
+
+    explainGatherCandidate(info, candidates);
+
+    return;
   }
 
-  trimVisibleCandidates(info, mostApplicable, visibleFns);
+  std::set<BlockStmt*> visited;
+  std::vector<BlockStmt*> currentScopes, nextScopes;
```

Comment:
I am hoping that this part of the code make the algorithm clear. But it is not so clear how the iteration starts. Could you add a comment about that if it's not easy to adjust the code itself?

In particular it would make more sense to me if the code near `// We are just starting out. Prime it with the scope of 'call'.` would move to this loop.

Reply:
Done, see Line 4397.